### PR TITLE
Fix assertion check to be PHP7+ compatible, fixes #381

### LINKED ISSUE #381: PHP 7.X =>Error triggered: assert(): Calling assert() with a string argument is deprecated
Changing to PHP 7.X (X=1, 2 or 3) will throw an error on the "Games tab".
---
Error triggered: assert(): Calling assert() with a string argument is deprecated

Error: "assert(): Calling assert() with a string argument is deprecated"
Raised: "/usr/www/users/vdiplo/vDiplomacy/gamesearch/searchItems.php"
Line: "271"
---
As PHP 7.0 is really old now webhosters are removing support for this version. Might be a good time to work on these funktions to make them future-proof.

### PR DESCRIPTION
**Pull request by splittingred**
> 7.0.0 | assert() is now a language construct and not a function. assertion can now be an expression. The second parameter is now interpreted either as an exception (if a Throwable object is given), or as the description supported from PHP 5.4.8 onwards.

https://www.php.net/manual/en/function.assert.php

Also fixes some reference errors with Countable that show up in PHP 7.2+.

### CODE REVIEW

**Comment by jmo1121109** [comment]
@Sleepcap can you take a look at this and see if this solves the issues you were having with php higher than 7 on vDip?

**Comment by diptobi1** [comment]
This does not fix all Countable reference errors yet, does it? (https://github.com/kestasjk/webDiplomacy/issues/656)

**Comment by splittingred** [comment]
@diptobi1 No, not all of them. That's a larger PR.

**Comment by splittingred** [comment]
Honestly the best approach here would be to merge #655, then add [phpcs](https://github.com/squizlabs/PHP_CodeSniffer) + [PHPCompatibility](https://github.com/PHPCompatibility/PHPCompatibility) to dev-require to check for PHP 7.2 support automatically.

**Comment by splittingred** [comment]
@jmo1121109 @shdant113 Can I have a review?

File: map/drawMap.php [hunk 1]
```diff
@@ -341,17 +341,16 @@ protected function loadOrderArrows()
 	{
 		foreach($this->orderArrows as &$orderArrow)
 		{
-			foreach($orderArrow as $name => &$param)
-			{
-				if ( $name == 'headAngle' ) $param = M_PI/$param;
+			foreach($orderArrow as $name => &$param) {
```
Comment (shdant113) [inline map/drawMap.php:344]: Can each of the opening and closing brackets within this loop live on their own line to conform with rest of the style throughout the codebase?

**Comment by splittingred** [comment]
@shdant113 Done, thanks!

**Comment by jmo1121109** [comment]
@kestasjk ended up taking a number of these, and then making the rest of the needed changes for php 7+ when he upgraded the server, so closing this one out.

**Pull request closed without merging**